tls/s2n_connection.c

```diff
@@ -467,6 +471,7 @@ int s2n_connection_free(struct s2n_connection *conn)
     s2n_x509_validator_wipe(&conn->x509_validator);
     GUARD(s2n_client_hello_free(&conn->client_hello));
     GUARD(s2n_free(&conn->application_protocols_overridden));
+    GUARD(s2n_stuffer_free(&conn->cookie_stuffer));
```

Comment:
add to `s2n_connection_free_handshake` as well?

the cookie doesn't need to be persisted after the connection is negotiated, so it can be freed earlier.

Reply:
Added to the `s2n_connection_free_handshake`.